Lib/pathlib.py

```diff
@@ -421,6 +260,54 @@ def __reduce__(self):
         # when pickling related paths.
         return (self.__class__, tuple(self._parts))
 
+    @classmethod
+    def _parse_parts(cls, parts):
+        parsed = []
+        sep = cls._flavour.sep
+        altsep = cls._flavour.altsep
+        drv = root = ''
+        it = reversed(parts)
+        for part in it:
+            if not part:
+                continue
+            if altsep:
+                part = part.replace(altsep, sep)
+            drv, rel = cls._flavour.splitdrive(part)
+            # According to POSIX path resolution:
+            # http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap04.html#tag_04_11
+            # "A pathname that begins with two successive slashes may be
+            # interpreted in an implementation-defined manner, although more
+            # than two leading slashes shall be treated as a single slash".
+            if cls._flavour is posixpath and rel[:2] == sep * 2 and rel[2:3] != sep:
```

Comment:
Is it dangerous to hard-code this logic based on the exact flavour? It's the only piece that's isn't abstracted in this method around `cls._flavour`.

Reply:
This is the trickiest bit of the whole PR I think. 

1. We could use `cls._flavour.normpath()` to normalise the root, but it also collapses `../` segments, and we don't want that behaviour in pathlib. We'd need to stitch together the correct path to make it work.
2. We could add a *keep_pardir* argument to `normpath()` (see #88482), but there's a C implementation that would also need adjustment, and it feels like a can of worms.
3. We could add something new to `os.path` like `normroot()`, or perhaps `root` and `altroot`. But they'd be pretty niche!

There are probably options I'm missing. Not sure of the best way forward here - what do you reckon?

Reply:
I've made `_parse_parts()` call `normpath()` only on the root, which seems to do the trick :-)